tests/test_token_indexer.py

```diff
@@ -0,0 +1,90 @@
+"""Tests for the PretrainedBertIndexer module."""
+import unittest
+import torch
+
+from allennlp.data.tokenizers import WordTokenizer
+from gector.tokenizer_indexer import PretrainedBertIndexer
+from allennlp.data.tokenizers.word_splitter import BertBasicWordSplitter
+from allennlp.data.vocabulary import Vocabulary
+
+
+class TestPretrainedTransformerIndexer(unittest.TestCase):
+    """
+    A test case that tests PretrainedBertIndexer methods.
+
+    Attributes:
+    tokens : List[Token]
+        List of AllenNLP Token objects
+
+    vocab : Vocabulary object
+        Vocabularies from pretrained transformers
+
+    model_name : str
+        Pre-trained model name
+
+    """
+
+    def setUp(self):
+        """Iniital setup."""
+        tokenizer = WordTokenizer(word_splitter=BertBasicWordSplitter())
+        sentence = "the Quick brown fox jumped over the laziest lazy elmo"
+        vocab_path = "../data/output_vocabulary"
+        self.tokens = tokenizer.tokenize(sentence)
+        self.vocab = Vocabulary.from_files(vocab_path)
+        self.model_name = 'roberta-base'
+     
+    def test_do_lowercase(self):
+        """Test tokenizer to handle setting do_lowercase to be True"""
+        token_indexer = PretrainedBertIndexer(pretrained_model=self.model_name, max_pieces_per_token=5,
+                                              do_lowercase=True, max_pieces=512,
+                                              special_tokens_fix=1)
+        indexed_tokens = token_indexer.tokens_to_indices(self.tokens, self.vocab, self.model_name)
+        assert indexed_tokens['bert'] == [627, 2119, 6219, 23602, 4262, 81, 5, 40154, 7098, 22414, 1615, 4992]
```

Comment:
It's usually better to use the `unitest.TestCase` methods since you are using a test class. 
```suggestion
        self.assertEqual(indexed_tokens['bert'], [627, 2119, 6219, 23602, 4262, 81, 5, 40154, 7098, 22414, 1615, 4992])
```

Please change all the others in the tests.